saw-core/src/Verifier/SAW/Term/Functor.hs

```diff
@@ -609,3 +609,7 @@ looseVars (Unshared f) = freesTermF (fmap looseVars f)
 -- | Compute the value of the smallest variable in the term, if any.
 smallestFreeVar :: Term -> Maybe Int
 smallestFreeVar = smallestBitSetElem . looseVars
+
+-- | Test whether a 'Term' is closed, i.e., has no free variables
+termIsClosed :: Term -> Bool
+termIsClosed t = looseVars t == emptyBitSet
```

Comment:
Nice, I've wanted a function like this for some time. It might be worth considering replacing all uses of `\t -> looseVars t == emptyBitSet` with `termIsClosed` throughout the codebase in a separate commit. (`git grep -n "looseVars.*== emptyBitSet" .` reveals many uses of this idiom.)